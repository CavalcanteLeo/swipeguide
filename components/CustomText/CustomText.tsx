import React from 'react';
import { Text, TextStyle } from 'react-native';

type TextType = 'title' | 'subtitle' | 'body' | 'caption';

interface CustomTextProps {
  children: React.ReactNode;
  type?: TextType;
  style?: TextStyle | TextStyle[];
  numberOfLines?: number;
  testID?: string;
}

const CustomText: React.FC<CustomTextProps> = ({
  children,
  type = 'body',
  style = {},
  numberOfLines = 0,
  testID,
  ...props
}) => {
  const styles = {
    titleText: {
      color: '#333',
      fontSize: 24,
      lineHeight: 28,
      letterSpacing: 0.1,
    },
    subtitleText: {
      color: '#333',
      fontSize: 20,
      lineHeight: 24,
      letterSpacing: 0.1,
    },
    bodyText: {
      color: '#000',
      fontSize: 17,
      lineHeight: 20,
      letterSpacing: 0.1,
    },
    captionText: {
      color: '#000',
      fontSize: 14,
      lineHeight: 18,
      letterSpacing: 0.1,
    },
  };

  const textStyles = {
    title: styles.titleText,
    subtitle: styles.subtitleText,
    body: styles.bodyText,
    caption: styles.captionText,
  };

  return (
    <Text
      style={[textStyles[type], style]}
      numberOfLines={numberOfLines}
      {...props}
      testID={testID}
    >
      {children}
    </Text>
  );
};

export default CustomText;
