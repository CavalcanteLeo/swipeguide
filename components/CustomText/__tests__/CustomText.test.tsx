import React from 'react';
import { render } from '@testing-library/react-native';
import CustomText from '../CustomText';

describe('CustomText', () => {
  it('renders the text with the correct type style', () => {
    const { getByText } = render(
      <CustomText type="title" testID="custom-text">
        Test Title
      </CustomText>
    );

    const textComponent = getByText('Test Title');
    expect(textComponent.props.style).toContainEqual({
      color: '#333',
      fontSize: 24,
      lineHeight: 28,
      letterSpacing: 0.1,
    });
  });

  it('renders the text with default body style when no type is specified', () => {
    const { getByText } = render(
      <CustomText testID="custom-text">Test Body</CustomText>
    );

    const textComponent = getByText('Test Body');
    expect(textComponent.props.style).toContainEqual({
      color: '#000',
      fontSize: 17,
      lineHeight: 20,
      letterSpacing: 0.1,
    });
  });

  it('applies additional styles passed via the style prop', () => {
    const additionalStyle = { color: 'red' };
    const { getByText } = render(
      <CustomText style={additionalStyle} testID="custom-text">
        Styled Text
      </CustomText>
    );

    const textComponent = getByText('Styled Text');
    expect(textComponent.props.style).toContainEqual(additionalStyle);
  });
});
