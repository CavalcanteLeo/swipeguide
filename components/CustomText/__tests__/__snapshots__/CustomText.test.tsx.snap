// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText respects the numberOfLines prop 1`] = `
<Text
  numberOfLines={2}
  style={
    [
      {
        "color": "#000",
        "fontSize": 17,
        "letterSpacing": 0.1,
        "lineHeight": 20,
      },
      {},
    ]
  }
  testID="custom-text"
>
  Long long long text, Long long long text, Long long long text, Long long long text, Long long long text
</Text>
`;
